symengine/pow.cpp

```diff
@@ -164,6 +164,20 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         RCP<const Pow> A = rcp_static_cast<const Pow>(a);
         return pow(A->get_base(), mul(A->get_exp(), b));
     }
+    if (eq(*a, *E)) {
+        if (is_a<Mul>(*b)
+            and is_a_Complex(*down_cast<const Mul &>(*b).get_coef())) {
+            const map_basic_basic &dict = down_cast<const Mul &>(*b).get_dict();
+            if (dict.size() == 1) {
+                for (const auto &p : dict) {
+                    if (eq(*p.first, *pi) and eq(*p.second, *one)) {
+                        return add(cos(mul(mul(I, minus_one), b)),
+                                   mul(I, sin(mul(mul(I, minus_one), b))));
```

Comment:
This seems correct, as you can always do this transformation. The only question remaining is whether this transformation is only applied when it can simplify the exp(), and it seems it is.

It is probably possible to optimize this return statement further, using the information we know (i.e. not calling sin/cos at all), but that can be done later.